concent_api/conductor/tasks.py

```diff
@@ -0,0 +1,76 @@
+#from celery import shared_task
+
+from golem_messages     import message
+
+from .models            import VerificationRequest
+from .models            import UploadReport
+from .models            import UploadRequest
+
+
+#@shared_task
+def verification_order_task(
+    subtask_id:             str,
+    src_code:               str,
+    extra_data:             str,
+    short_description:      str,
+    working_directory:      str,
+    performance:            str,
+    docker_images:          str,
+    source_file:            str,
+    result_file:            str,
+):
+    pass
+
+
+#@shared_task
+def verification_request_task(
+    compute_task_def:   message.ComputeTaskDef,
+    files:              list,
+):
+    assert isinstance(compute_task_def, message.ComputeTaskDef)
+
+    # The app creates a new instance of VerificationRequest.
+    verification_request = VerificationRequest(
+        subtask_id          = compute_task_def['subtask_id'],
+        src_code            = compute_task_def['src_code'],
+        extra_data          = compute_task_def['extra_data'],
+        short_description   = compute_task_def['short_description'],
+        working_directory   = compute_task_def['working_directory'],
+        performance         = compute_task_def['performance'],
+        docker_images       = compute_task_def['docker_images'],
+    )
+    verification_request.full_clean()
+    verification_request.save()
+
+    # The app creates ne UploadRequest for each file listed in the task.
+    for file in files:
+        upload_request = UploadRequest(
+            path                 = file,
+            verification_request = verification_request,
+        )
+        upload_request.full_clean()
+        upload_request.save()
+
+        # If there are already UploadReports corresponding to some files, the app links them together by setting the
+        # value of the foreign key in UploadRequest.
+        UploadReport.objects.filter(
+            path = file
```

Comment:
You should link only those reports that are not already associated with an `UploadRequest`.

Reply:
:+1: Ok, that was not defined exactly in description.

Reply:
Yeah, it wasn't, though it's hard to describe it completely down to that level of detail and I try not to unless the detail is important.

The behavior in case when the file has already been uploaded in not well defined because this should not happen under normal circumstances. But it's technically possible here. I think that if, for whatever reason, `UploadRequest` is already associated with a file uploaded earlier we should not overwrite it. Either leave it as is or raise an error.

Reply:
I will leave it as is by excluding them from query.